### PR TITLE
Prevent "Cant guess python-indent-offset..." messages in tests.

### PR DESCRIPTION
**Pull request by trittweiler**
```
* eglot-tests.el (eglot--tests--python-mode-bindings): New. Sets
`python-indent-guess-indent-offset-verbose' to nil.
(auto-detect-running-server)
(auto-shutdown)
(auto-reconnect)
(basic-diagnostics)
(rename-a-symbol)
(basic-completion)
(basic-xref)
(snippet-completions)
(snippet-completions-with-company)
(hover-after-completions)
(python-autopep-formatting)
(python-yaps-formatting)
(eglot-ensure)
(slow-sync-connection-wait)
(slow-sync-connection-intime)
(slow-async-connection)
(slow-sync-timeout): Use `eglot--tests--python-mode-bindings' in these tests.
```

### CODE REVIEW

File: eglot-tests.el [hunk 1]
```diff
@@ -232,6 +232,14 @@ Pass TIMEOUT to `eglot--with-timeout'."
          (eglot-connect-timeout timeout))
     (apply #'eglot--connect (eglot--guess-contact))))
 
+(defvar eglot--tests--python-mode-bindings
+  '(;; Prevent "Can't guess python-indent-offset ..." messages.
+    (python-indent-guess-indent-offset-verbose . nil))
+  "Alist of variable bindings to be used in tests involving `python-mode'.")
+
+
```
Comment (trittweiler) [inline eglot-tests.el:240]: (Here's actually a Linefeed (^L))

File: eglot-tests.el [hunk 2]
```diff
@@ -232,6 +232,14 @@ Pass TIMEOUT to `eglot--with-timeout'."
          (eglot-connect-timeout timeout))
     (apply #'eglot--connect (eglot--guess-contact))))
 
+(defvar eglot--tests--python-mode-bindings
```
Comment (joaotavora) [inline eglot-tests.el:235]: Do you foresee a lot of such bindings for `python-mode`?  If only one or two I wonder if plugging it directly in the `eglot--with-fixture` macro wouldn't be enough.  I know it breaks an abstraction layer, but I'd do it. Other than that, very clean :-)

Comment (trittweiler) [reply eglot-tests.el:235]: Heh. YAGNI vs. "Broken Window Theory".

Can you approve this, or do you want me to change it?

Comment (joaotavora) [reply eglot-tests.el:235]: Sure, let's merge it, no problem.  If I'm ever in that file and I find it's too much, I might do the simpler thing (and if you ever find yourself needing a lot of python specific vars, we revisit this).

**Pull request merged**